### ISSUE TITLE
Chi function crashes

### ISSUE THREAD

**Issue by faridhassani** [scqubits/scqubits#88]
[fluxonium-chi-calculation.zip](https://github.com/scqubits/scqubits/files/5574876/fluxonium-chi-calculation.zip)
Hi,
I am trying to plot the dispersive shifts of a fluxonium qubit, however the "scqubits.utils.sweep_plotting.chi" crashes with an error of " 'ParameterSweep' object has no attribute 'chi' ". the line of code that I'm running is after constructing the interaction term and successfully interacting with "explorer". I'm running:

import scqubits.utils.sweep_plotting as splot
splot.chi(sweep)

I also tried with splot.chi(explorer), but got the same thing. I will attach the code for further investigations. Let me know if I'm simply using the function wrong.
Best,
Farid.

**Comment by jkochNU** [comment]
This part of the library needs improvements and better documentation. 
`ParameterSweep` is not expected to build all quantities / observable that could be of interest automatically (so to mitigate runtime). Dispersive shifts thus have to be generated manually before they can be displayed.

Assuming `sweep` is your `ParameterSweep` instance, this should work:
```
import scqubits.core.sweep_generators as swp
chi_data_dict = swp.generate_chi_sweep(sweep)

chi_sweep = chi_data_dict[(<<N>>,<<M>>)]
splot.chi_01(chi_sweep)
```

Here, `<<N>>` and `<<M>>` are to be replaced by the indices of your oscillator subsystem and your qubit subsystem.

**Comment by faridhassani** [comment]
Thanks Jens,

It worked just as you instructed. The only question I have is that now the "splot.chi(chi_sweep)" should also work but I get an index out of range error as in the following:

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-345-d8ff9fa9bd3f> in <module>
      3 
      4 chi_sweep = chi_data_dict[(1,0)]
----> 5 splot.chi(chi_sweep)

~\AppData\Local\Continuum\Anaconda3\lib\site-packages\scqubits\utils\sweep_plotting.py in chi(datastore, **kwargs)
    149     ydata = datastore.chi
    150     xdata = datastore.param_vals
--> 151     state_count = ydata.shape[1]
    152     label_list = list(range(state_count))
    153     return plot.data_vs_paramvals(xdata, ydata, label_list=label_list, **defaults.chi(datastore.param_name, **kwargs))

IndexError: tuple index out of range

**Comment by jkochNU** [comment]
@faridhassani `ParameterSweep` and dispersive shifts have undergone some changes in the release of v2. (See docs and new example notebook.) Any feedback would be appreciated.

**Comment by faridhassani** [comment]
Just watched your talk in March meeting. V2 looks amazing I'll definitely update and come back to you on possible issues. The app for fitting spectroscopy sounded awesome. Currently for me is a back and forth between Mathematica and scqubits which because of the nice file system is not that big of a issue but still having it in one place is amazing.

**Comment by faridhassani** [comment]
Just checked the dispersive and lamb shifts in V2 of scqubits and I didn't find any issue at moment and I could fit experimental data just fine.

**Comment by jkochNU** [comment]
Addressed with release of version 2.0.
